File: download_stj.py
"""
Baixa PDFs dos Informativos STJ para envio ao servidor.
Execute na raiz do projeto: python download_stj.py
Os PDFs ficam em stj_pdfs/ e devem ser enviados para o servidor com scp.
"""
import requests
import os
import time

session = requests.Session()
session.headers.update({
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
})

os.makedirs("stj_pdfs", exist_ok=True)

START = 780
END = 900

print(f"Baixando informativos STJ edições {START}–{END}...")
print()

baixados = 0
for edition in range(START, END + 1):
    pdf_path = f"stj_pdfs/Informativo_{edition:04d}.pdf"
    if os.path.exists(pdf_path):
        print(f"[ok] Edição {edition:04d} já existe, pulando.")
        baixados += 1
        continue

    url = f"https://processo.stj.jus.br/SCON/GetPDFINFJ?edicao={edition:04d}"
    try:
        r = session.get(url, timeout=30)
        if r.ok and r.content[:4] == b"%PDF":
            with open(pdf_path, "wb") as f:
                f.write(r.content)
            print(f"[ok] Edição {edition:04d} baixada ({len(r.content) // 1024} KB)")
            baixados += 1
        else:
            print(f"[--] Edição {edition:04d} não encontrada (HTTP {r.status_code})")
    except Exception as e:
        print(f"[!!] Edição {edition:04d} erro: {e}")

    time.sleep(1)

print()
print(f"Concluído: {baixados} PDFs em stj_pdfs/")
print()
print("Para enviar ao servidor:")
print("  scp stj_pdfs/*.pdf root@2.24.75.193:/opt/judicore/_internal/data/stj_informativos/docs/")

File: download_stj.mjs
import { writeFileSync, existsSync, mkdirSync } from "fs";
import { join } from "path";

const START = 780;
const END = 900;
const OUT_DIR = "stj_pdfs";
const BASE_URL = "https://processo.stj.jus.br/SCON/GetPDFINFJ?edicao=";

mkdirSync(OUT_DIR, { recursive: true });

console.log(`Baixando informativos STJ edições ${START}–${END}...\n`);

let baixados = 0;

for (let edition = START; edition <= END; edition++) {
  const num = String(edition).padStart(4, "0");
  const pdf_path = join(OUT_DIR, `Informativo_${num}.pdf`);

  if (existsSync(pdf_path)) {
    console.log(`[ok] Edição ${num} já existe, pulando.`);
    baixados++;
    continue;
  }

  const url = `${BASE_URL}${num}`;
  try {
    const res = await fetch(url, {
      headers: {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36",
      },
    });

    if (res.ok) {
      const buf = Buffer.from(await res.arrayBuffer());
      if (buf.slice(0, 4).toString() === "%PDF") {
        writeFileSync(pdf_path, buf);
        console.log(`[ok] Edição ${num} baixada (${Math.round(buf.length / 1024)} KB)`);
        baixados++;
      } else {
        console.log(`[--] Edição ${num} não é PDF (HTTP ${res.status})`);
      }
    } else {
      console.log(`[--] Edição ${num} não encontrada (HTTP ${res.status})`);
    }
  } catch (e) {
    console.log(`[!!] Edição ${num} erro: ${e.message}`);
  }

  await new Promise((r) => setTimeout(r, 1000));
}

console.log(`\nConcluído: ${baixados} PDFs em ${OUT_DIR}/`);
console.log("\nPara enviar ao servidor:");
console.log(`  scp ${OUT_DIR}/*.pdf root@2.24.75.193:/opt/judicore/_internal/data/stj_informativos/docs/`);
